GVFS/GVFS.Common/Enlistment.cs

```diff
@@ -90,5 +90,10 @@ public virtual GitProcess CreateGitProcess()
         {
             return new GitProcess(this);
         }
+
+        public void ReuseExistingAuth(GitAuthentication authentication)
+        {
+            this.Authentication = authentication;

```

Comment:
I made this a method because I wanted to make it a bigger hammer than just setting a property. But the right fix is to just plumb the auth instance around so we can pass it directly to the `GVFSEnlistment` constructor and not have to swap it out afterwards. I'll make that change.